### PR TITLE
support for TLS client certificates

### PR DESCRIPTION
**Pull request by weboo**

### CODE REVIEW

File: cmd/mqtt2prometheus.go [hunk 1]
```diff
@@ -182,3 +189,27 @@ func setupExtractor(cfg config.Config) (metrics.Extractor, error) {
...
+func newTlsConfig(cfg config.Config) *tls.Config {
+	certpool := x509.NewCertPool()
+	pemCerts, err := ioutil.ReadFile(cfg.MQTT.CACert)
+	if err == nil {
+		certpool.AppendCertsFromPEM(pemCerts)
+	}
+
+	cert, err := tls.LoadX509KeyPair(cfg.MQTT.ClientCert, cfg.MQTT.ClientKey)
+	if err != nil {
+		return &tls.Config{}
```
Comment (hikhvar) [inline cmd/mqtt2prometheus.go:202]: Could you please return an error here? A panic is very user unfriendly.

File: cmd/mqtt2prometheus.go [hunk 2]
```diff
@@ -182,3 +189,27 @@ func setupExtractor(cfg config.Config) (metrics.Extractor, error) {
 	}
 	return nil, fmt.Errorf("no extractor configured")
 }
+
+func newTlsConfig(cfg config.Config) *tls.Config {
+	certpool := x509.NewCertPool()
+	pemCerts, err := ioutil.ReadFile(cfg.MQTT.CACert)
+	if err == nil {
```
Comment (hikhvar) [inline cmd/mqtt2prometheus.go:196]: Could you please return an error here? Silently ignoring the error will lead to lot of user frustration while debugging the own setup.

**Review by hikhvar — CHANGES_REQUESTED** [review]
Could you in general return an error if the TLS setup is invalid? And please make a non-TLS setup still viable.

File: cmd/mqtt2prometheus.go (hunk 1 above)
Comment (weboo) [reply cmd/mqtt2prometheus.go]: I've modified to return errors. thanks.

File: cmd/mqtt2prometheus.go (hunk 2 above)
Comment (weboo) [reply cmd/mqtt2prometheus.go:201]: As for here, CA cert may not be required.
In that case, you can catch it in a later process.

Comment (hikhvar) [reply cmd/mqtt2prometheus.go:201]: Yes but the user might specified an invalid path. E.g. a typo. In that case you will just silently ignore the error and the user wonders why the TLS configuration does not work.

Comment (weboo) [reply cmd/mqtt2prometheus.go]: You're right. I've revised to return error.
If it does not exists CACert settings, skip here.

**Review by hikhvar — APPROVED** [review]

**Pull request merged**